controllers/topics/index.js

```diff
@@ -218,12 +247,18 @@ const followTopicV2 = async (req, res) => {
   }
 };
 
-const _afterPutTopic = async (isUnfollow, token, userId, name) => {
+const _afterPutTopic = async (isUnfollow, token, userId, name, isAnonymous) => {
   // follow / unfollow main feed topic
-  if (isUnfollow) {
-    await unfollowMainFeedTopic(token, userId, name);
-  } else {
-    await followMainFeedTopic(token, userId, name);
+  try {
+    if (isUnfollow) {
+      await unfollowMainFeedTopic(token, userId, name);
+      await removeTopicFromChatTab(token, name, userId);
+    } else {
+      await followMainFeedTopic(token, userId, name);
+      await addTopicToChatTab(token, name, userId, isAnonymous);
+    }
+  } catch (error) {
+    console.log('After put topic error: ', error);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling here is not sufficient. Simply logging the error to the console might not be enough for production-level code. Consider throwing the error or handling it in a way that would be appropriate for your application's error handling strategy. Here is a suggestion:

```diff
-    console.log('After put topic error: ', error);
+    throw error;
```
This will allow the calling function to handle the error as needed, which could include logging the error, retrying the operation, or informing the user of the issue.


<!-- This is an auto-generated comment by OSS CodeRabbit -->